src/Raven.Server/Documents/Queries/FieldsToFetch.cs

```diff
@@ -258,6 +258,22 @@ private static void ThrowInvalidFetchAllStoredDocuments()
                 result[key] = val;
             }
 
+            if (metadata.HasHighlightings)
```

Comment:
Can we support highlights without projecting the field?

Reply:
This adds a highlighted field to be also returned in projection (even if not requested). Effectively this will also change the behavior of Lucene queries (previously we didn't return the field in projection, now it will).

Can we retrieve that field during query execution (since we need the value to handle highlighting) but skip returning it on the projection?

Reply:
This is when explicitly required:
![image](https://github.com/ravendb/ravendb/assets/1045411/a3b1450c-79e3-4cba-9ee9-782be4fc7311)
This is when not:
![image](https://github.com/ravendb/ravendb/assets/1045411/d81b50cb-1d91-4ec7-8ba7-f85860d887f1)

How did you trigger the difference because `QueryMetadata.HasHighlightings` should be `false` if we are not asking for including the highlightings.

Reply:
The issue is that even if we _don't_ request `Content` field in projection we will still return it. The following query:

```
from index 'MultiSearch' select id() as Id, Title, Url, Category include highlight(Content,128,1,$p0)
{"p0":{"PreTags":["<span class='search-result-highlight'>"],"PostTags":["</span>"]}}
```

didn't return `Content` field previously:

![image](https://github.com/ravendb/ravendb/assets/376551/20eb05bd-f49f-4127-912a-f41d0bc919bf)

but with this change it will return it (on Lucene and Corax):

![image](https://github.com/ravendb/ravendb/assets/376551/6362c8e2-a7a9-4d5d-979b-e9a7f5032df2)


So my question is: can we retrieve that field internally for Corax in order to do highlighting but do not return in in projection results since it was not requested?